module/zfs/zpl_file.c

```diff
@@ -361,8 +361,10 @@ zpl_write(struct file *filp, const char __user *buf, size_t len, loff_t *ppos)
 	ssize_t wrote;
 
 	crhold(cr);
+	spl_inode_lock(file_inode(filp));
 	wrote = zpl_write_common(filp->f_mapping->host, buf, len, ppos,
 	    UIO_USERSPACE, filp->f_flags, cr);
+	spl_inode_unlock(file_inode(filp));
```

Comment:
> For refrence check the generic_file_write_iter does it that way

This is not exactly right. In generic_file_write_iter the sync part is outside the lock.